### PR TITLE
[Lang] Support GPU solve with analyzePattern and factorize

### PR DESCRIPTION
**Pull request by FantasyVR**
Issue: #2906

### CODE REVIEW

File: taichi/program/sparse_solver.cpp [hunk 1]
```diff
@@ -249,6 +315,30 @@ void CuSparseSolver::solve_cu(Program *prog,
...
+  // Retrive the info of the sparse matrix
+  SparseMatrix *sm_no_cv = const_cast<SparseMatrix *>(&sm);
+  CuSparseMatrix *A = dynamic_cast<CuSparseMatrix *>(sm_no_cv);
+  size_t rowsA = A->num_rows();
+  size_t d_b = prog->get_ndarray_data_ptr_as_int(&b);
+  size_t d_x = prog->get_ndarray_data_ptr_as_int(&x);
+  CUSOLVERDriver::get_instance().csSpScsrcholSolve(
+      cusolver_handle_, rowsA, (void *)d_b, (void *)d_x, info_, gpu_buffer_);
+
+  // TODO: free allocated memory and handles
```
Comment (strongoier) [inline taichi/program/sparse_solver.cpp:332]: nit: are these related to this PR?

**Review by strongoier — APPROVED** [review]

File: taichi/program/sparse_solver.cpp (hunk 1 above)
Comment (FantasyVR) [reply taichi/program/sparse_solver.cpp:332]: Yeah. The allocated memory and handles should not be freed here, but they are allocated in this function.

**Review by Hanke98 — APPROVED** [review]
LGTM!
